### ISSUE TITLE
Updates for supporting clusters in Azure

### ISSUE THREAD

**Issue by yardbirdsax** [PowerShell/xFailOverCluster#29]
From what I can see, this module would need a few tweaks in order to work properly in the context of creating failover clusters in Azure. For example, when creating the cluster you have to assign a link-local IP address as the cluster IP, then bring it back online. I'd love to contribute towards this but thought I should inquire as to the best method? For example, should these be created as separate resources in the same module (xClusterAzure?), or even a fork of the entire repository (xFailoverClusterAzure?)?

**Comment by sujitks** [comment]
Hi Guys, Does this mean xFailoverCluster is not supported in Azure VM?

**Comment by glennmate** [comment]
Yes, we need the ability to point the cluster to the primary node for joining, not the CNO object (default behavior of the powershell DSC address).  This is the same issue referenced here:

**Comment by glennmate** [comment]
Hey @johlju, Is there an ETA on getting xFailoverCluster functional for Azure? This is a huge feature gap / limitation.

**Comment by johlju** [comment]
@glennmate There is nobody in the community working on this yet, as far as I know. So I'm afraid until somebody is working on it, there is no ETA.

@yardbirdsax Sorry for not answering on this one. Couldn't it be the same resource, but a parameter saying if it should create a link-local IP? Or do you see other problems that would make the need to make an entire new resource? 

To all: What is missing from the resources in this repo for being able to create a cluster in Azure?

**Comment by yardbirdsax** [comment]
I think the changes are significant enough here that it should be a separate resource. For example, not only do you have to create a link-local IP (I guess technically you don't, but unless you put an Azure load balancer in place I don't think the assigned IP will work at all, so it's a waste of space), but you also have to have all commands run against the local node, not the CNO.

If it'd be helpful I could put in the finished code I have that, at least as far as I can tell, is working. I don't have tests for everything (shame on me, I know :) ), but it'd at least be a start.

**Comment by johlju** [comment]
@yardbirdsax please send in the PR. I might be able to help out with the unit tests, but might take a few weeks before I can get around to it. Would you be up to fixing other changes that might come up in the review?

**Comment by yardbirdsax** [comment]
As long as it's okay that it may be a few weeks, I can probably work towards any comments that come up. I'll collect the code and put the PR in.

**Comment by johlju** [comment]
Thats perfectly fine :) And at any time let me know and I will help out any way I can.

**Comment by StandBackBurrito** [comment]
I have seen multiple issues and PRs for this, but nothing completed (or at least from what I saw)

Is there a workaround I can use to get clusters setup in azure with DSC?

It has been a while since any work has been done on this package, is it dead? If so is there a replacement FailOverCluster DSC package for this?

**Comment by johlju** [comment]
The repository is still being released if there are any new merged changes. There are no one in the community currently working on any PRs, nor have I seen any fixes around this.

**Comment by johlju** [comment]
There is another module created by @rchaganti that maybe supports this, see
https://github.com/rchaganti/FailoverClusterDSC
